jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -376,8 +376,15 @@ private boolean parseRequestBuffer()
     @Override
     public void onCompleted()
     {
+        // If we are fill interested, then a read is pending and we must abort
+        if (isFillInterested())
+        {
+            LOG.warn("Pending read in onCompleted {} {}", this, getEndPoint());
+            abort(new IllegalStateException());
+        }
```

Comment:
This is the main fix for the read side.  Nothing should be fillInterested at completion, so we abort if there is, which will stop us accepting future requests.